src/topology/metric_space/isometric_smul.lean

```diff
@@ -0,0 +1,287 @@
+/-
+Copyright (c) 2022 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import topology.metric_space.isometry
+
+/-!
+# Group actions by isometries
+
+In this file we define two typeclasses:
+
+- `has_isometric_smul M X` says that `M` multiplicatively acts on an (pseudo extended) metric space
+  `X` by isometries;
+- `has_isometric_vadd` is an additive version of `has_isometric_smul`.
+
+We also prove basic facts about isometric actions and define bundled isometries
+`isometric.const_mul`, `isometric.mul_left`, `isometric.mul_right`, `isometric.div_left`,
+`isometric.div_right`, and `isometric.inv`, as well as their additive versions.
+-/
```

Comment:
I think this should include some discussion about the meaning of `has_isometric_smul Gᵐᵒᵖ G` and `has_isometric_smul G G`.